data-prepper-plugins/aggregate-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/aggregate/AggregateProcessorIT.java

```diff
@@ -502,6 +511,51 @@ void aggregateWithHistogramAggregateAction() throws InterruptedException, NoSuch
         }
     }
 
+    @ParameterizedTest
+    @ValueSource(doubles = {20.0, 40.0, 60.0})
```

Comment:
Is this really testing the sampling?

Since you are using the error condition, it seems that is taking precedence.

I think you may be able to create a test for the sampling. This could then have assertions on the size based on the sampling configuration.

Reply:
Yes. it is testing sampling but not exactly to the mentioned numbers because of the "probabilistic sampling" there is no way to test it exactly. The error condition is applied only to some events. I could only test for `lessThan` as the exact number is never guaranteed with probabilistic sampling.

Reply:
Yea, I was thinking about that as well. Do you think this would be flaky if we had an assertion that tested that it was within 10 percentage points each way?